pkg/controller/servicebindingrequest/secret.go

```diff
@@ -52,21 +54,61 @@ func (s *Secret) createOrUpdate(payload map[string][]byte) (*unstructured.Unstru
 	ns := s.plan.Ns
 	name := s.plan.Name
 	logger := s.logger.WithValues("Namespace", ns, "Name", name)
-	secretObj := &corev1.Secret{
-		ObjectMeta: metav1.ObjectMeta{
-			Namespace: ns,
-			Name:      name,
-		},
-		Data: payload,
+
+	var bindingRef interface{}
+	resource := SecretResource
+
+	if s.plan.SBR.Spec.BindingReference == nil {
+		return nil, err.New("BindingReference is empty")
```

Comment:
If ` bindingRef == nil,` a binding secret should be created , `resource = SecretResource` . It should not pass errors, IMO

Reply:
Done, added a test too https://github.com/redhat-developer/service-binding-operator/pull/366/commits/c8566ef6c187e26e9e987ffe184925e1e10b4fe0